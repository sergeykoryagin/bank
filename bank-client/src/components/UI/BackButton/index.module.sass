@import "../../../styles/mixins"

.backButton
    width: 48px
    height: 48px

.backIcon
    width: 100%
    height: 100%
    fill: $color-blue-300