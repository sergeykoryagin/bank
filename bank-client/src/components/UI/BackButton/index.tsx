import { ReactComponent as ArrowBack } from 'assets/svg/arrow-back.svg';
import cn from 'classnames';
import styles from 'components/UI/BackButton/index.module.sass';
import { VFC } from 'react';
import { useNavigate } from 'react-router-dom';

interface Props {
    className?: string;
    to: string;
}

export const BackButton: VFC<Props> = ({ className, to }: Props) => {
    const navigate = useNavigate();
    const handleBackClick = () => navigate(to);
    return (
        <button className={cn(styles.backButton, className)} onClick={handleBackClick}>
            <ArrowBack className={styles.backIcon} />
        </button>
    );
};
