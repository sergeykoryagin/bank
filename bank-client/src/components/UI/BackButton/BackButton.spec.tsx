import { fireEvent, render, screen } from '@testing-library/react';
import { BackButton } from 'components/UI/BackButton/index';
import { ReactElement } from 'react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';

const renderWithRouter = (component: ReactElement) =>
    render(<MemoryRouter>{component}</MemoryRouter>);

const TestApp = () => {
    return (
        <div>
            <BackButton href='/menu' />
            <Routes>
                <Route path='/' element={<div>home page</div>} />
                <Route path='/menu' element={<div>menu page</div>} />
            </Routes>
        </div>
    );
};

describe('BackButton component', () => {
    it('should renders correctly', () => {
        const { container } = renderWithRouter(<BackButton href='' />);
        expect(screen.getByRole('button')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it('should navigate after click', () => {
        renderWithRouter(<TestApp />);
        expect(screen.getByText(/home page/i)).toBeInTheDocument();
        expect(screen.queryByText(/menu page/i)).not.toBeInTheDocument();

        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();

        fireEvent.click(button);

        expect(screen.queryByText(/home page/i)).not.toBeInTheDocument();
        expect(screen.getByText(/menu page/i)).toBeInTheDocument();
    });
});
