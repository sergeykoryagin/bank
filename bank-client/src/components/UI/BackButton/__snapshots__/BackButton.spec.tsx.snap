// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component should renders correctly 1`] = `
<div>
  <button
    class="backButton"
  >
    <svg
      class="backIcon"
    >
      arrow-back.svg
    </svg>
  </button>
</div>
`;
